sales_team_security/models/res_partner.py

```diff
@@ -11,6 +13,64 @@ class ResPartner(models.Model):
 
     # add indexes for better performance on record rules
     user_id = fields.Many2one(index=True)
+    team_id = fields.Many2one(
+        "crm.team",
+        string="Sales Team",
+        compute="_compute_team_id",
+        precompute=True,  # avoid queries post-create
+        ondelete="set null",
+        readonly=False,
+        store=True,
+        index=True,
+    )
+
+    @api.model
+    def get_view(self, view_id=None, view_type="form", **options):
+        """
+        Patch view to inject the default value for the team_id and user_id.
+        """
+        # FIXME: Use base_view_inheritance_extension when available
+        res = super().get_view(view_id, view_type, **options)
+        if view_type == "form":
+            eview = etree.fromstring(res["arch"])
+            xml_fields = eview.xpath("//field[@name='child_ids']")
+            if xml_fields:
+                context_str = (
+                    xml_fields[0]
+                    .get("context", "{}")
+                    .replace(
+                        "{",
+                        "{'default_team_id': team_id, 'default_user_id': user_id,",
+                        1,
+                    )
+                )
+                xml_fields[0].set("context", context_str)
+            res["arch"] = etree.tostring(eview)
+        return res
+
+    @api.onchange("parent_id")
+    def _onchange_parent_id_sales_team_security(self):
+        """If assigning a parent partner and the contact doesn't have
+        team or salesman, we put the parent's one (if any).
+        """
+        if self.parent_id and self.parent_id.team_id and not self.team_id:
+            self.team_id = self.parent_id.team_id.id
+        if self.parent_id and self.parent_id.user_id and not self.user_id:
+            self.user_id = self.parent_id.user_id.id
```

Comment:
IMHO, this is totally out of scope. Moreover this is already done by: https://github.com/odoo/odoo/blob/18.0/odoo/addons/base/models/res_partner.py#L397

Those changes should be done in computes.